plugin-infra/go-plugin-access/test/com/thoughtworks/go/plugin/access/elastic/ElasticAgentExtensionTest.java

```diff
@@ -101,4 +102,28 @@ public void getCapabilitiesIsSupportedOnlyInVersion2() throws Exception {
 
         verify(pluginManager, times(0)).submitTo(any(), any());
     }
+
+    @Test
+    public void shouldSerializeServerInfoToJSONFor_V1() throws Exception {
+        String pluginId = "plugin_id";
+        ElasticAgentExtension extension = new ElasticAgentExtension(pluginManager);
+
+        when(pluginManager.resolveExtensionVersion(pluginId, extension.goSupportedVersions())).thenReturn("1.0");
+
+        String serverInfoJSON = extension.serverInfoJSON(pluginId, "x12adf", "http://my.build.com", "https://my.build.com");
+
+        assertThat(serverInfoJSON, Is.is("{\"server_id\":\"x12adf\",\"site_url\":\"http://my.build.com\",\"secure_site_url\":\"https://my.build.com\"}"));
+    }
+
+    @Test
+    public void shouldSerializeServerInfoToJSONFor_V2() throws Exception {
```

Comment:
Move this to ElasticAgentExtensionTestForV2 to be consistent with NotificationExtensionTestForV* ? Or the other way around?